qiskit/assembler/assemble_schedules.py

```diff
@@ -53,15 +54,30 @@ def assemble_schedules(schedules, qobj_id, qobj_header, run_config):
     meas_lo_range = qobj_config.pop('meas_lo_range', None)
     meas_map = qobj_config.pop('meas_map', None)
 
-    max_memory_slot = 0
-
     instruction_converter = instruction_converter(PulseQobjInstruction, **qobj_config)
 
     lo_converter = LoConfigConverter(PulseQobjExperimentConfig,
                                      qubit_lo_range=qubit_lo_range,
                                      meas_lo_range=meas_lo_range,
                                      **qobj_config)
 
+    # Find maximum number of memory slot and validate acquires
+    max_memory_slot = 0
+    for schedule in schedules:
```

Comment:
this is good point. to enable zero padding, we need to add `memory_slots` to each experimental header. experimental header is generated in each loop, so we need to know `memory_slots` before the loop starts. this is why I added another loop (filter) before that.

Reply:
another solution is to assume `n_qubit == memory_slots` (I think this is general case). because the experimental header of `QuantumCircuit` contains `n_qubit`, we can replace the loop to find `memory_slots` by `n_qubit`.

Reply:
circuit assembler receives `creg_sizes` and just need to sum up. but in pulse assembler, we have no `creg_sizes` provided. I'll keep "track the max memory_slot per-experiment" strategy, and generate pulse experiment qobj separately after the loop. I think this works.

But, what I don't understand is why we can set independent `memory_slots` for each experiment? Currently `memory_slots` in experimental header (not one in pulse qobj config) seems to be used just for zero padding.

When a job contains two schedules as follows
- schedule1 contains memory slot 0, 1, 2
- schedule2 contains memory slot 0, 1, 2, 3, 4, 5

should we set `memory_slots=3` for schedule1 and `memory_slots=6` for schedules? This seems to be intended in current header configuration. If so, the routine to decide `memory_slots` will be totally different.